src/shared/protocol.test.ts

```diff
@@ -62,6 +62,182 @@ describe("protocol tests", () => {
     await transport.close();
     expect(oncloseMock).toHaveBeenCalled();
   });
+
+  test("should reset timeout when progress notification is received", async () => {
+    jest.useFakeTimers();
```

Comment:
Can you please put this into a `beforeEach` and `useRealTimers` into an `afterEach`? (Consider using a `describe` to group with the below tests too.)

Right now, errors in the test could cause timers not to be restored.